### PR TITLE
Add assistiveTechnologyEnabled flag to window

### PR DESCRIPTION
**Pull request by jonahwilliams**
Used to distinguish between only having an a11y service enabled on Android (lastpass?) and actually using talkback or other assistive tech.  iOS mapping is the same as semantics enabled, except disabled for the simulator.

### CODE REVIEW

File: lib/ui/window.dart [hunk 1]
```diff
@@ -637,6 +637,30 @@ class Window {
   bool get semanticsEnabled => _semanticsEnabled;
   bool _semanticsEnabled = false;
 
+  /// Whether the user is using assitive technologies to interact with the
+  /// application.
+  /// 
+  /// This includes screen readers such as TalkBack on Android and VoiceOVer
+  /// on iOS, as well as hardware switches, and more.
+  /// 
+  /// THe [onAssistiveTechnologyEnabled] callback is called whenever this value
```
Comment (goderbauer) [inline lib/ui/window.dart]: THe -> The

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 2]
```diff
@@ -812,11 +815,14 @@ public void onTouchExplorationStateChanged(boolean enabled) {
             if (enabled) {
                 mTouchExplorationEnabled = true;
                 ensureAccessibilityEnabled();
+                nativeSetAssistiveTechnologyEnabled(mNativeView.get(), true);
             } else {
                 mTouchExplorationEnabled = false;
                 if (mAccessibilityNodeProvider != null) {
                     mAccessibilityNodeProvider.handleTouchExplorationExit();
                 }
+                nativeSetSemanticsEnabled(mNativeView.get(), false);
```
Comment (goderbauer) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Where is this set to true? Kinda expected that in the if branch?

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 3]
```diff
@@ -792,16 +792,19 @@ - (NSUInteger)supportedInterfaceOrientations {
...
   // inspector is enabled on the simulator. We conservatively always turn on the
-  // accessibility bridge in the simulator.
-  bool enabled = true;
+  // accessibility bridge in the simulator, but never assistive technology.
+  platformView->SetSemanticsEnabled(true);
+  platformView->SetAssistiveTechnologyEnabled(false);
 #else
   bool enabled = UIAccessibilityIsVoiceOverRunning() || UIAccessibilityIsSwitchControlRunning() ||
-                 UIAccessibilityIsSpeakScreenEnabled();
+                UIAccessibilityIsSpeakScreenEnabled();
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Should "SpeakScreen" set assitiveTechnologiesEnabled to true? This one feels kinda borderline...

File: shell/platform/android/io/flutter/view/FlutterView.java (hunk 2 above)
Comment (jonahwilliams) [reply shell/platform/android/io/flutter/view/FlutterView.java]: That is in ensureAccessibilityEnabled

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 3 above)
Comment (jonahwilliams) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Fixed?

File: lib/ui/window.dart (hunk 1 above)
Comment (jonahwilliams) [reply lib/ui/window.dart]: Fixed

**Pull request merged**